### ISSUE TITLE
Support AVA's assertion methods

### ISSUE THREAD

**Issue by TomerAberbach** [dubzzz/ava-fast-check#10]
Thanks for creating `fast-check`. It's an amazing property-based testing framework!

I've been using `ava-fast-check`, and it works well, but I think the integration with AVA would be better if instead of being limited to a predicate function, you could use [AVA's assertions from the `t` parameter](https://github.com/avajs/ava/blob/master/docs/03-assertions.md) like so:
```js
import { testProp, fc } from 'ava-fast-check';

// `t` is from ava
testProp('should detect the substring', [fc.string(), fc.string(), fc.string()], (t, a, b, c) => {
  t.true((a + b + c).includes(b));
});
```
Of course, this is kind of a trivial example, but in general when I choose between writing normal and property-based tests (I write both) I feel like I also have to choose between two different test-writing styles (expressive ava-style and simple predicate).

I also understand that this would be a breaking change, but I think it's worth it. I would be willing to make a PR for this so let me know your thoughts!

**Comment by dubzzz** [comment]
Thanks a lot for the suggestion.

It seems to be an idiomatic way to use `ava` so yes I believe this is a really great idea even if it should introduce a breaking change (I will bump the major for `ava-fast-check` in that case).

> I would be willing to make a PR for this so let me know your thoughts!

Waiting for the PR!

**Comment by dubzzz** [comment]
**Just a quick question as I am not a user of  `ava`, I was wondering the following:**

Is `ava-fast-check` simplifying the use of `fast-check` on `ava`?
For the moment, I only covered `ava` but I should maybe cover others if there is a huge benefit for such wrappers.

**Comment by TomerAberbach** [comment]
> Thanks a lot for the suggestion.
> 
> It seems to be an idiomatic way to use `ava` so yes I believe this is a really great idea even if it should introduce a breaking change (I will bump the major for `ava-fast-check` in that case).
> 
> > I would be willing to make a PR for this so let me know your thoughts!
> 
> Waiting for the PR!

Great! I'll try to get to it this week.

> **Just a quick question as I am not a user of `ava`, I was wondering the following:**
> 
> Is `ava-fast-check` simplifying the use of `fast-check` on `ava`?
> For the moment, I only covered `ava` but I should maybe cover others if there is a huge benefit for such wrappers.

I think `ava-fast-check` simplifies the use of `fast-check` in `ava` in a couple of ways:
- It reduces boilerplate. I don't have to write a `fast-check` "wrapper" around every test like so:
  ```js
  test('my test', async t => {
    await fc.assert(fc.asyncProperty(arb1, arb2, (value1, value2) => {
      // My actual test code
    }))
  })
  ```
- It makes tests that use `fast-check` look more like normal `ava` tests. I feel I have a good handle on how `fast-check` works, but I didn't always and not everyone who works on a project with me necessarily does. Making the tests look more "normal" makes it easier for developers to contribute (I think most developers haven't encountered property-based testing).

And I think it makes it easier for new users to pick up `fast-check` in a couple of ways:
- It lowers the barrier to entry for property-based testing if you already know how to use `ava`.
- It indicates very clearly that `fast-check` works with `ava`. In the JavaScript ecosystem there are so many tools, and developers always have to figure out which of them work together. Seeing a package called `ava-fast-check` immediately communicates to developers that `fast-check` works with `ava`, and that it's going to be easy to use them together.

**Comment by dubzzz** [comment]
Thanks a lot for the feedback. It confirms that specific wrappers can be a plus, I should continue my work on `jest-fast-check` too.
